allennlp/tests/common/file_utils_test.py

```diff
@@ -97,6 +101,43 @@ def test_url_to_filename_with_etags_eliminates_quotes(self):
             assert back_to_url == url
             assert etag == "mytag"
 
+    def test_split_s3_path(self):
+        # Test splitting good urls.
+        assert split_s3_path("s3://my-bucket/subdir/file.txt") == ("my-bucket", "subdir/file.txt")
+        assert split_s3_path("s3://my-bucket/file.txt") == ("my-bucket", "file.txt")
+
+        # Test splitting bad urls.
+        with pytest.raises(ValueError):
+            split_s3_path("s3://")
+            split_s3_path("s3://myfile.txt")
+            split_s3_path("myfile.txt")
+
+    def test_s3_request_decorator(self):
+        fake_url = "s3://my-bucket/myfile.txt"
+
+        # Good request.
+        @s3_request
+        def good_s3_request(url):
+            return url
+
+        assert good_s3_request(fake_url) == fake_url
+
+        # Request that fails because the file is missing.
+        @s3_request
+        def missing_file_request(url):
+            raise ClientError({"Error": {"Code": 404}}, "s3_request to {}".format(url))
+
+        with pytest.raises(FileNotFoundError):
+            missing_file_request(fake_url)
+
+        # Request that fails due to other reasons.
+        @s3_request
+        def other_bad_request(url):
+            raise ClientError({"Error": {"Code": 503}}, "s3_request to {}".format(url))
+
+        with pytest.raises(ClientError):
+            other_bad_request(fake_url)
+
```

Comment:
I have not used boto much, is there a way of testing it that uses a mock s3 to test the actual calls, or is that not done?

Reply:
Yea there might be a better way of doing this, like here: https://stackoverflow.com/questions/37143597/mocking-boto3-s3-client-method-python. I'll look into it and see if I can improve those tests

Reply:
if it's a ton of work it's probably not worth it, but if it's relatively easy I think it's a good test to have